packages/parser-sass/tests/interface-tests.ts

```diff
@@ -0,0 +1,165 @@
+    const { engine, processor } = mockContext();
+
+    const engineEmitAsyncSpy = sandbox.spy(engine, 'emitAsync');
+    const postcssProcessSpy = sandbox.spy(processor, 'process');
+    const elementGetAttributeSpy = sandbox.spy(element, 'getAttribute');
+
+    await engine.emitAsync('element::style', { element, resource } as ElementFound);
+
+    t.true(elementGetAttributeSpy.calledOnce);
+    t.is(elementGetAttributeSpy.firstCall.args[0], 'lang');
+    t.true(postcssProcessSpy.calledOnce);
+    t.is(postcssProcessSpy.firstCall.args[0], code);
+    t.true(engineEmitAsyncSpy.calledThrice);
+
+    const args = engineEmitAsyncSpy.args[2];
+    const data = args[1] as StyleParse;
+
+    t.is(args[0], 'parse::end::css');
+    t.is(data.code, code);
+    t.is(data.element, element);
+    t.is(data.resource, resource);
+});
+
+test('If fetch::end::* is received, then we should check if it is SASS, parse the stylesheet, and emit a parse::end::css event', async (t) => {
+    const sandbox = sinon.createSandbox();
+    const { engine, processor } = mockContext();
+    const code = `
+.foo
+    color: #fff
+`;
+
+    const engineEmitAsyncSpy = sandbox.spy(engine, 'emitAsync');
+    const postcssProcessSpy = sandbox.spy(processor, 'process');
+
+    await engine.emitAsync('fetch::end::unknown', {
+        resource: 'styles.sass',
+        response: {
+            body: { content: code },
+            mediaType: 'text/x-sass'
+        }
+    } as FetchEnd);
+
+    t.true(postcssProcessSpy.calledOnce);
+    t.is(postcssProcessSpy.firstCall.args[0], code);
+    t.true(engineEmitAsyncSpy.calledThrice);
+
+    const args = engineEmitAsyncSpy.thirdCall.args;
+    const data = args[1] as StyleParse;
+
+    t.is(args[0], 'parse::end::css');
+    t.is(data.code, code);
+    t.is(data.element, null);
+    t.is(data.resource, 'styles.sass');
+});
+
+test('If fetch::end::* is received, then we should check if it is SCSS, parse the stylesheet, and emit a parse::end::css event', async (t) => {
+    const sandbox = sinon.createSandbox();
+    const { engine, processor } = mockContext();
+    const code = '.foo { color: #fff }';
+
+    const engineEmitAsyncSpy = sandbox.spy(engine, 'emitAsync');
+    const postcssProcessSpy = sandbox.spy(processor, 'process');
+
+    await engine.emitAsync('fetch::end::unknown', {
+        resource: 'styles.scss',
+        response: {
+            body: { content: code },
+            mediaType: 'text/x-scss'
+        }
+    } as FetchEnd);
+
+    t.true(postcssProcessSpy.calledOnce);
+    t.is(postcssProcessSpy.firstCall.args[0], code);
+    t.true(engineEmitAsyncSpy.calledThrice);
+
+    const args = engineEmitAsyncSpy.thirdCall.args;
+    const data = args[1] as StyleParse;
+
+    t.is(args[0], 'parse::end::css');
+    t.is(data.code, code);
+    t.is(data.element, null);
+    t.is(data.resource, 'styles.scss');
+});
+
+test('If fetch::end::* is received for CSS, nothing should happen', async (t) => {
```

Comment:
What about the following? It took me a second to understand what the intention was:

```suggestion
test('If fetch::end::css is emitted, nothing should happen', async (t) => {
```